ffi/derive/src/impl_visitor.rs

```diff
@@ -131,18 +134,30 @@ impl<'ast> ImplDescriptor<'ast> {
 }
 
 impl FnDescriptor {
-    fn from_impl_method(self_ty: &syn::Path, node: &syn::ImplItemMethod) -> Self {
-        let mut visitor = FnVisitor::new(Some(self_ty));
+    pub(crate) fn from_impl_method(
+        self_ty: &syn::Path,
+        trait_name: Option<&syn::Path>,
+        node: &syn::ImplItemMethod,
+    ) -> Self {
+        let mut visitor = FnVisitor::new(Some(self_ty), trait_name);
 
         visitor.visit_impl_item_method(node);
         FnDescriptor::from_visitor(visitor)
     }
 
+    pub(crate) fn from_fn(node: &syn::ItemFn) -> Self {
+        let mut visitor = FnVisitor::new(None, None);
+
+        visitor.visit_item_fn(node);
+        Self::from_visitor(visitor)
+    }
+
     fn from_visitor(visitor: FnVisitor) -> Self {
         Self {
             self_ty: visitor.self_ty.map(Clone::clone),
+            trait_name: visitor.trait_name.map(Clone::clone),
 
-            doc: visitor.doc.expect_or_abort("Missing doc"),
+            doc: visitor.doc,
```

Comment:
We need a way to handle docs properly. Most trait implementations don't have doc-comments, true, but they need to inherit the docs from the trait. Since we carried the docs for ordinary functions through, I suggest doing this in another ticket.

Reply:
We don't carry docs from functions, we replace docs to something like: 
```rust
/// FFI function equivalent of [`Struct::method`] (or [`Trait::method`])
```

IMO having link to original method is enough in this case.

My suggestion is that @mversic add docs here to later use them when generating opaque wrapper structs.